tests/test_sparse.py

```diff
@@ -0,0 +1,53 @@
+import random
+
+import jax.numpy as jnp
+import pytest
+
+import dynamiqs as dq
+
+
+class TestSparseDIA:
+    @pytest.fixture(autouse=True)
+    def _setup(self):
+        N = 4
+        self.matrixA = dq.destroy(N)
+        self.matrixB = dq.number(N)
+        self.sparseA = dq.to_sparse(self.matrixA)
+        self.sparseB = dq.to_sparse(self.matrixB)
+
+    def test_add(self, rtol=1e-05, atol=1e-08):
+        out_dia_dia = (self.sparseA + self.sparseB).to_dense()
+        out_dia_dense = self.sparseA + self.matrixB
+        out_dense_dia = self.matrixA + self.sparseB
+        out_dense_dense = self.matrixA + self.matrixB
+
+        assert jnp.allclose(out_dense_dense, out_dia_dia, rtol=rtol, atol=atol)
+        assert jnp.allclose(out_dense_dense, out_dia_dense, rtol=rtol, atol=atol)
+        assert jnp.allclose(out_dense_dense, out_dense_dia, rtol=rtol, atol=atol)
+
+    def test_sub(self, rtol=1e-05, atol=1e-08):
+        out_dia_dia = (self.sparseA - self.sparseB).to_dense()
+        out_dia_dense = self.sparseA - self.matrixB
+        out_dense_dia = self.matrixA - self.sparseB
+        out_dense_dense = self.matrixA - self.matrixB
+
+        assert jnp.allclose(out_dense_dense, out_dia_dia, rtol=rtol, atol=atol)
+        assert jnp.allclose(out_dense_dense, out_dia_dense, rtol=rtol, atol=atol)
+        assert jnp.allclose(out_dense_dense, out_dense_dia, rtol=rtol, atol=atol)
+
+    def test_transform(self, rtol=1e-05, atol=1e-08):
```

Comment:
```suggestion
    def test_convert(self, rtol=1e-05, atol=1e-08):
```
(this test function should appear either after `test_mul` or before `test_add`)